pkg/storage/localdir.go

```diff
@@ -0,0 +1,72 @@
+package storage
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+	"github.com/operator-framework/operator-registry/alpha/property"
+)
+
+// LocalDir is a storage Instance. When Storing a new FBC contained in
+// fs.FS, the content is first written to a temporary file, after which
+// it is copied to it's final destination in RootDir/catalogName/. This is
+// done so that clients accessing the content stored in RootDir/catalogName have
+// atomic view of the content for a catalog.
+type LocalDir struct {
+	RootDir string
+}
+
+func (s LocalDir) Store(catalog string, fsys fs.FS) error {
+	tmpDir := filepath.Join(s.RootDir, catalog+"-temp")
+	if err := os.MkdirAll(tmpDir, 0700); err != nil {
+		return err
+	}
+	defer os.RemoveAll(tmpDir)
+	fbcDir := filepath.Join(s.RootDir, catalog)
+	if err := os.MkdirAll(fbcDir, 0700); err != nil {
+		return err
+	}
+	tempFile, err := os.OpenFile(filepath.Join(tmpDir, "all.json"), os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
```

Comment:
Use `os.Create` here? We never expect this file to already exist in the tmpDir right?
```suggestion
	tempFile, err := os.Create(filepath.Join(tmpDir, "all.json"), 0644)
```

Reply:
This is called on update as well as the original store.